File: src/rclpy/xxxx/xxxx/mytalker.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class MyTalker(Node):
    def __init__(self):
        super().__init__("mytalker_node")
        self.pub_ = self.create_publisher(String, "mytopic", 10)
        self.timer = self.create_timer(0.5, self.timer_callback)
        self.counter = 0

    def timer_callback(self):
        msg = String()
        msg.data = "Hello, world!"
        self.pub_.publish(msg)
        self.get_logger().info("Publishing: '%s'" % msg.data)
        self.counter += 1


def main(args=None):
    print("main")
    rclpy.init(args=args)
    mytalker_node = MyTalker()
    try:
        rclpy.spin(mytalker_node)
    except KeyboardInterrupt:
        pass
    finally:
        mytalker_node.destroy_timer(mytalker_node.timer)
        mytalker_node.destroy_node()
        rclpy.shutdown()


if __name__ == "__main__":
    print("hello ros2")
    main()

File: src/rclcpp/tut1/src/mytalker.cpp
// write a simple node which mimic the talker node
// Please use ROS2 stack to implement this node
//
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using namespace std::chrono_literals;

class MyTalker : public rclcpp::Node {
  public:
    MyTalker() : Node("my_talker") {
        publisher_ = this->create_publisher<std_msgs::msg::String>("chatter", 10);
        timer_ = this->create_wall_timer(50ms, std::bind(&MyTalker::timer_callback, this));
    }

  private:
    void timer_callback() {
        auto message = std_msgs::msg::String();
        message.data = "Hello, world! " + std::to_string(count_++);
        RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
        publisher_->publish(message);
    }
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    size_t count_;
};

int main(int argc, char* argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MyTalker>());
    rclcpp::shutdown();
    return 0;
}
